src/main/java/com/google/edith/servlets/UserInsights.java

```diff
@@ -0,0 +1,198 @@
+    */
+  public Map<String, Double> aggregateUserData() { 
+    Entity userStats = retreiveUserStats();
+    List<Key> items = (List<Key>) userStats.getProperty("Items");
+
+    items.sort(SORT_BY_DATE);
+
+    return calculateWeeklyTotal(items);
+  }
+
+  /**
+   * Creates a String-Intger map relating weekly periods to spending.
+   * @param items - A list of {@code Key} objects that reference Item entities
+                  - in the datastore.
+   * @return Creates a map with keys for each ending day of a weekly period and  
+   *         values for the total spending during that period. 
+   */
+  public Map<String, Double> calculateWeeklyTotal(List<Key> items)  {
+    Map<String, Double> weeklyTotals = new HashMap<String, Double>();
+    LocalDate currentEndOfWeek;
+    try {
+      currentEndOfWeek = getEndOfWeek(LocalDate.parse((String) datastore.get(items.get(0))
+                                          .getProperty("date"), dateFormatter));
+    } catch(EntityNotFoundException e) {
+      System.err.println("Error: Entity could not be located");
+      return null;
+    }
+    double weeklyTotal = 0;
+    Entity itemEntity;
+    for(Key item : items) {
+      try {
+        itemEntity = datastore.get(item);
+        LocalDate itemDate = LocalDate.parse((String) datastore.get(item).getProperty("date"),
+                                             dateFormatter);
+        if(ChronoUnit.DAYS.between(itemDate, currentEndOfWeek) < 0) {
+          weeklyTotals.put(currentEndOfWeek.toString(), weeklyTotal);
+          currentEndOfWeek = getEndOfWeek(itemDate);
+          weeklyTotal = 0;
+        }
+
+        weeklyTotal += ((double) itemEntity.getProperty("price")) * 
+                           ((long) itemEntity.getProperty("quantity"));
+    
+      } catch (EntityNotFoundException e) {
+        System.err.println("Error: Entity could not be found");
+      } 
+    }
+    
+    weeklyTotals.put(currentEndOfWeek.toString(), weeklyTotal);
+
+    return weeklyTotals;
+  }
+
+  /**
+   * Finds the Date for the last day of the week that {@code itemDate} is in 
+   * (weeks start on Monday and end on Sunday).
+   * @param itemDate - Uses the int day value of the week from this 
+   *                   paramter to calculate the end of its week.
+   * @return the LocalDate representing the end of the week {@code itemDate}
+   *         is in.
+   */
+  public LocalDate getEndOfWeek(LocalDate itemDate) {
+    int currentDayOfWeek = itemDate.getDayOfWeek().getValue();
+    return itemDate.plusDays(7 - currentDayOfWeek);
+  }  
+  
+  /**
+   * Creates a Json string that contains the weekly aggregate for this user
+   * and the items this user purchased.
+   * @return a Json formatted String of items and an aggregate.
+   */
+  public String createJson() {
+    Map<String, String> aggregateValues = aggregateUserData().entrySet().stream()
+    .collect(Collectors.toMap(Map.Entry::getKey, entry -> Double.toString(entry.getValue())));
```

Comment:
I think this part should be indented more

Reply:
Done.

Reply:
Oh this is from an older commit. I added more in between .stream() and .collect().